### PR TITLE
Perplexity Hotfix

### PR DESCRIPTION
**Pull request by cle-ros**
Hotfix for perplexity, because it requires special aggregation behavior

### CODE REVIEW

File: flambe/metric/dev/perplexity.py [hunk 1]
```diff
@@ -28,3 +30,42 @@ def compute(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
...
+
+        Returns
+        -------
+        dict
+            the state dict
+        """
+        pred, target = args
+        if not state:
+            state['historic_score'] = []
+        state['historic_score'].append(self.entropy(pred, target).cpu().detach())
```
Comment (jeremyasapp) [inline flambe/metric/dev/perplexity.py]: I think we can be more efficient by keeping only the sum:
```python
logits = self.entropy(pred, target).cpu().detach()
state['score'] += logits.sum()
state['num_samples'] += len(logits)
```
Then in finalize:
```python
logits = self.entropy(pred, target).cpu().detach()
return torch.exp(state['scores'] / state['num_samples']).item()
```

File: flambe/metric/dev/perplexity.py [hunk 2]
```diff
@@ -28,3 +30,42 @@ def compute(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
...
+
+        Parameters
+        ----------
+        state: dict
+            the metric state
+
+        Returns
+        -------
+        float
+            The final score.
```
Comment (jeremyasapp) [inline flambe/metric/dev/perplexity.py]: make sure to leave a blank line at the end of the docstring

File: flambe/metric/dev/perplexity.py (hunk 1 above)
Comment (cle-ros) [reply flambe/metric/dev/perplexity.py]: done

File: flambe/metric/dev/perplexity.py (hunk 2 above)
Comment (cle-ros) [reply flambe/metric/dev/perplexity.py]: done

**Comment by cle-ros** [comment]
all done;
also included more test cases
